apiserver/facades/agent/machine/machiner.go

```diff
@@ -82,18 +81,23 @@ func (api *MachinerAPI) SetMachineAddresses(args params.SetMachinesAddresses) (p
 			results.Results[i].Error = common.ServerError(common.ErrPerm)
 			continue
 		}
-		err = common.ErrPerm
-		if canModify(tag) {
-			var m *state.Machine
-			m, err = api.getMachine(tag)
-			if err == nil {
-				addresses := params.NetworkAddresses(arg.Addresses...)
-				err = m.SetMachineAddresses(addresses...)
-			} else if errors.IsNotFound(err) {
-				err = common.ErrPerm
-			}
+		if !canModify(tag) {
+			results.Results[i].Error = common.ServerError(common.ErrPerm)
+			continue
+		}
+		m, err := api.getMachine(tag)
+		if err != nil {
+			results.Results[i].Error = common.ServerError(err)
+			continue
+		}
+		addresses, err := params.ToProviderAddresses(arg.Addresses...).ToSpaceAddresses(api.st)
```

Comment:
Conversion via lookup now happens for all incoming `ProviderAddresses` that need storage with space IDs.